src/components/refactor/dialogs/generator/modification/generator-modification-dialog.js

```diff
@@ -0,0 +1,352 @@
+    PLANNED_OUTAGE_RATE,
+    Q_PERCENT,
+    RATED_NOMINAL_POWER,
+    REACTIVE_CAPABILITY_CURVE_CHOICE,
+    REACTIVE_CAPABILITY_CURVE_TABLE,
+    REACTIVE_POWER_SET_POINT,
+    STARTUP_COST,
+    TRANSFORMER_REACTANCE,
+    TRANSIENT_REACTANCE,
+    VOLTAGE_LEVEL,
+    VOLTAGE_REGULATION,
+    VOLTAGE_REGULATION_ON,
+    VOLTAGE_REGULATION_TYPE,
+    VOLTAGE_SET_POINT,
+} from '../../../utils/field-constants';
+
+import {
+    fetchEquipmentInfos,
+    modifyGenerator,
+} from '../../../../../utils/rest-api';
+import { sanitizeString } from '../../../../dialogs/dialogUtils';
+import { REGULATION_TYPES } from '../../../../network/constants';
+import GeneratorModificationForm from './generator-modification-form';
+import {
+    getSetPointsEmptyFormData,
+    getSetPointsSchema,
+} from '../set-points/set-points-utils';
+import {
+    getReactiveLimitsEmptyFormData,
+    getReactiveLimitsSchema,
+} from '../reactive-limits/reactive-limits-utils';
+import {
+    assignPreviousValuesToForm,
+    assignValuesToForm,
+    getPreviousValuesEmptyForm,
+    PREVIOUS_P,
+    PREVIOUS_Q_MAX_P,
+    PREVIOUS_Q_MIN_P,
+    PREVIOUS_VOLTAGE_REGULATION,
+} from './generator-modification-utils';
+import { useIntl } from 'react-intl';
+
+const schema = yup
+    .object()
+    .shape(
+        {
+            [EQUIPMENT_ID]: yup.string().nullable().required(),
+            [EQUIPMENT_NAME]: yup.string(),
+            [ENERGY_SOURCE]: yup.string().nullable(),
+            [MAXIMUM_ACTIVE_POWER]: yup.number().nullable(),
+            [MINIMUM_ACTIVE_POWER]: yup.number().nullable(),
+            [RATED_NOMINAL_POWER]: yup.number().nullable(),
+            [TRANSIENT_REACTANCE]: yup.number().nullable(),
+            [TRANSFORMER_REACTANCE]: yup.number().nullable(),
+            [PLANNED_ACTIVE_POWER_SET_POINT]: yup.number().nullable(),
+            [STARTUP_COST]: yup.number().nullable(),
+            [MARGINAL_COST]: yup.number().nullable(),
+            [PLANNED_OUTAGE_RATE]: yup
+                .number()
+                .nullable()
+                .min(0, 'RealPercentage')
+                .max(1, 'RealPercentage'),
+            [FORCED_OUTAGE_RATE]: yup
+                .number()
+                .nullable()
+                .min(0, 'RealPercentage')
+                .max(1, 'RealPercentage'),
+            ...getSetPointsSchema(true),
+            ...getReactiveLimitsSchema(true),
+        },
+        [MAXIMUM_REACTIVE_POWER, MINIMUM_REACTIVE_POWER]
+    )
+    .required();
+
+const GeneratorModificationDialog = ({
+    editData,
+    defaultIdValue,
+    currentNode,
+    studyUuid,
+    ...dialogProps
+}) => {
+    const currentNodeUuid = currentNode.id;
+    const { snackError } = useSnackMessage();
+    const intl = useIntl();
+
+    //in order to work properly, react hook form needs all fields to be set at least to null
+    const emptyFormData = useMemo(
+        () => ({
+            [EQUIPMENT_ID]: defaultIdValue ?? null,
```

Comment:
Remove defaultIdValue so when selecting id, validate button is enebled:
[EQUIPMENT_ID]: '',

And refactor emptyFormData as global constant